apollo-router/tests/integration/subscriptions/ws_passthrough.rs

```diff
@@ -917,3 +918,159 @@ async fn test_subscription_ws_passthrough_dedup() -> Result<(), BoxError> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_subscription_ws_passthrough_dedup_close_early() -> Result<(), BoxError> {
+    if !graph_os_enabled() {
+        eprintln!("test skipped");
+        return Ok(());
+    }
+
+    // Create fixed payloads for consistent testing
+    let custom_payloads = vec![create_user_data_payload(1), create_user_data_payload(2)];
+    let interval_ms = 50;
+    let is_closed = Arc::new(AtomicBool::new(false));
```

Comment:
this is for checking that the subscription channel being closed, right? maybe let's make it extra clear:
```suggestion
    let is_subscription_channel_closed = Arc::new(AtomicBool::new(false));
```